src/mainviews/create-job.tsx

```diff
@@ -288,6 +290,14 @@ export function CreateJob(props: ICreateJobProps): JSX.Element {
             environment={props.model.environment}
             value={props.model.outputFormats || []}
           />
```

Comment:
The environments promise should not be handled within the `EnvironmentPicker` if the network request is being called from the parent. `environmentsPromise` (which really should be renamed to `fetchEnvironments` as it is a function) is being invoked every render. Look what happens when I add a console log to the function body:

https://user-images.githubusercontent.com/44106031/192658491-cc8fa074-6dff-4cd6-9a72-8cf69258d16f.mov

You need to lift this state up and make the fetch environments response run only once on initial render like so:

```
const [envList, setEnvList] = useState<Scheduler.IRuntimeEnvironment[]>([]);

...

useEffect(async () => {
  setEnvList(await environmentsPromise());
}, []);
```

Then pass `envList` to both `EnvironmentPicker` and `ComputeTypePicker`.

Reply:
Thank you! Piyush was just asking about this.

Reply:
The `useEffect` function can't take an `async` function — I get an error when I try to use the code you provided. There is a workaround, but be warned: it's ugly even by JavaScript standards. https://stackoverflow.com/questions/56838392/how-to-call-an-async-function-inside-a-useeffect-in-react

Reply:
Oh yeah, forgot about that quirk of useEffect callbacks.